rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -1842,6 +1851,13 @@ private Optional<String> getRevisionFromCalledWorkflow(String calledWorkflow) {
         }
     }
 
+    public static Map<String, String> setSubmissionModeToGenericInfo(Map<String, String> genericInformation) {
+        Map<String, String> newGenericInformation = genericInformation == null ? new LinkedHashMap<>()
+                                                                               : new LinkedHashMap<>(genericInformation);
+        newGenericInformation.put("submission.mode", SCHEDULER_API);
```

Comment:
sorry, I forgot to mention "submission.mode" should also be defined as a global constant